src/Microsoft.HttpRepl.Tests/Commands/ListCommandTests.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.HttpRepl.Commands;
+using Microsoft.HttpRepl.Fakes;
+using Microsoft.HttpRepl.Preferences;
+using Microsoft.Repl.Parsing;
+using Xunit;
+
+namespace Microsoft.HttpRepl.Tests.Commands
+{
+    public class ListCommandTests : CommandTestsBase
+    {
+        [Fact]
+        public async Task ExecuteAsync_NoBaseAddress_ShowsWarning()
+        {
+            ArrangeInputs(commandText: "ls",
+                          baseAddress: "http://lcoalhost/",
+                          path: "/",
+                          urlsWithResponse: null,
+                          out MockedShellState shellState,
+                          out HttpState httpState,
+                          out ICoreParseResult parseResult,
+                          out _,
+                          out IPreferences preferences);
+
+            httpState.BaseAddress = null;
+            httpState.SwaggerEndpoint = new Uri("http://localhost/swagger.json");
+            httpState.Structure = new DirectoryStructure(null);
+
+            ListCommand listCommand = new ListCommand(preferences);
+
+            await listCommand.ExecuteAsync(shellState, httpState, parseResult, CancellationToken.None);
+
+            Assert.Single(shellState.Output);
+            Assert.Equal(Resources.Strings.ListCommand_Error_NoBaseAddress, shellState.Output[0]);
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_NoSwagger_ShowsWarning()
+        {
+            ArrangeInputs(commandText: "ls",
+                          baseAddress: "http://localhost/",
+                          path: "/",
+                          urlsWithResponse: null,
+                          out MockedShellState shellState,
+                          out HttpState httpState,
+                          out ICoreParseResult parseResult,
+                          out _,
+                          out IPreferences preferences);
+
+            httpState.BaseAddress = new Uri("http://localhost/swagger.json");
```

Comment:
set base to a swagger file in a test named NoSwagger is a little confusing.  Why isn't it just `http://localhost/`' here?  (And isn't that set by ArrangeInputs above?)

Reply:
Yeah, that shouldn't have been there. Same with the next test. It didn't change anything (it's just checking whether BaseAddress is set to anything) but it was confusing and unnecessary, so I've removed it.